lib/matplotlib/projections/polar.py

```diff
@@ -521,9 +523,17 @@ def set_thetagrids(self, angles, labels=None, frac=None, fmt=None,
         ACCEPTS: sequence of floats
         """
         # Make sure we take into account unitized data
-        angles = self.convert_yunits(angles)
-        angles = np.asarray(angles, np.float_)
-        self.set_xticks(angles * (np.pi / 180.0))
+        if isinstance( angles[0], float ):
+           # If the data is floats it is assumed to be degrees (as per the
+           # docsting for this method)
+           angles = np.asarray(angles, np.float_)
+           self.set_xticks(angles * (np.pi / 180.0))
+        else:
+           # The unit converters are defined to return radians
+           angles = self.convert_yunits(angles)
```

Comment:
ah, I think I know what you are getting at. Yes, radius is the considered the "y-axis", while theta is considered the "x-axis" in the sense that that is what you use for calling tickers and labels (so, ax.set_xlabels() for setting the theta labels). However, it appears that `convert_xunits()` has traditionally been used for radius and `convert_yunits()` have been traditionally used for theta. I don't know if that is a bug or not. Perhaps this never worked correctly?